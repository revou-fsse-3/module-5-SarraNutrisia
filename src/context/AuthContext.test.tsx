import { render, screen } from '@testing-library/react';
import { AuthProvider } from '../context/AuthContext';
import { useAuth } from '../context/AuthContext';

describe('AuthContext component unit testing', () => {
    
    it('should render AuthContext with children', () => {
        
        const ChildComponent = () => {
            const { isAuthenticated, login, logout } = useAuth();
            return (
                <div>
                    <p>{`isAuthenticated: ${isAuthenticated}`}</p>
                    <button onClick={() => login('token')}>Login</button>
                    <button onClick={() => logout()}>Logout</button>
                </div>
            );
        };

        render(
            <AuthProvider>
                <ChildComponent />
            </AuthProvider>
        );

        const isAuthenticatedElement = screen.getByText(/isAuthenticated/i);
        expect(isAuthenticatedElement).toMatchSnapshot();
    });
});