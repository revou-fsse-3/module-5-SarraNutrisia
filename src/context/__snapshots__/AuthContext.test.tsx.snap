// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthContext component unit testing should render AuthContext with children 1`] = `
<p>
  isAuthenticated: false
</p>
`;
